### DISCUSSION TITLE
Program runs few lines then Checks and Freezes

### DISCUSSION THREAD

**Discussion by Rsawmill** [arkypita/LaserGRBL#2040] [General]
Has anyone experienced this? GRBL program runs for a few lines. Then goes into a self check? Then locks up. Have to hit reset to get it to  go back home or do anything. If I try to rerun burn I have to reload the image. I am including a picture of the g code screen. 
I am still learning g code and all that LaserGRBL can do. Any help would be appreciated. THank you,R.

![PXL_20230319_210039730](https://user-images.githubusercontent.com/46383491/226461692-45ee1c8e-b2b5-4e0e-a1c8-21cc114ac517.jpg)
![PXL_20230319_205217224](https://user-images.githubusercontent.com/46383491/226461908-f17a8ff7-f73f-430c-bf90-01c8b83bdc91.jpg)

**Comment by arkypita** [comment]
Your machine does not respond as a standard grbl machine
In details, LaserGRBL is not receiving the expected "ok\r" messages back from the machine, instead he receive those weird {1/1}ok that is a non standard and not expected message. I suggest to check specification of your machine to see what firmware is your machine using, because it's not a grbl one

**Reply by Rsawmill** [reply]
How can I find out what firmware . When I bought the laser in 2019 from the "two win" store on aliexpress they said in their description that it would run GRBL. It came with CNCC2019. I never could figure it out, so I have been running Lightburn since.  Lightburn has worked well for this LED laser for cutting and plain engraving. I just can't get grayscale to come out on LB and my other LED laser that ran GRBL, control boards died. Thank you.R.

**Reply by arkypita** [reply]
I can give you some help to figure out why your machine is not working with lasergrbl, but you should give me some info following this instruction.

1. Update your lasergrbl to latest version (latest is v4.9.4, your is 3.0.9)
2. With the machine not connected go to menu "?" and select "activate extended log". This feature log on file the whole communication between lasergrbl and the machine.
3. Connect to the machine
4. Load any image and send it to the machine (by press play).
5. After some seconds (about 20s) abort the job and disconnect the machine
6. Send the recorded log file to me at info@lasergrbl.com (or upload it here)

Here is how the comlog look like

![image](https://user-images.githubusercontent.com/8782035/226872404-89e5db6a-9076-4064-9301-2e9c32e1ad91.png)


One other file that could help me is the standard LaserGRBL session log available under "?", "Open Session Log"

Here is how session log look like:

![image](https://user-images.githubusercontent.com/8782035/226873078-b281cc0f-f3dc-4201-bed0-1c2ce6545d5b.png)

**Reply by Rsawmill** [reply]
I sent email with the Logs.

**Reply by arkypita** [reply]
What I see is that until 20/01/2023 your machine was detected as Grbl v1.1f, and was working.
Then since 19/03/2023 your machine is detected as Grbl v0.8 

What happened? Are you using a different machine? Did you have to change the control board? Did you reprogram the firmware?
Grbl 0.8 is a very old firmware, I have never/few tested LaserGrbl with this old version.

However, even if old, basic functions should be compatible.

**Reply by arkypita** [reply]
The issue is all with the "{0/0}ok\r" message, that is not the standard "ok\r" message that grbl should send back to acknowledge a command.

**Comment by Rsawmill** [comment]
Arkypita, It's the same board and machine. As far as Firmware, I don't know what happened. I don't know how it could have changed. I don't even know how to change/flash the firmware.How do I reflash the firmware and where can I get the programming to reflash it? I'm not real electronics/tech savy. Thank you for your help, Ron.

    On Friday, March 24, 2023 at 01:56:40 PM PDT, arkypita ***@***.***> wrote:  
 
 


What I see is that until 20/01/2023 your machine was detected as Grbl v1.1f, and was working.
Then since 19/03/2023 your machine is detected as Grbl v0.8

What happened? Are you using a different machine? Did you have to change the control board? Did you reprogram the firmware?
Grbl 0.8 is a very old firmware, I have never/few tested LaserGrbl with this old version.

However, even if old, basic functions should be compatible.

**Reply by arkypita** [reply]
LaserGrbl offer the possibility to flash the latest grbl firmware from "tool" menu. 
You may need to re-configure the machine after flash.

**Reply by arkypita** [reply]
https://lasergrbl.com/usage/flash/

**Reply by arkypita** [reply]
you can find some guides on youtube and google
